packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -294,12 +294,13 @@ export class Cluster extends Resource implements ICluster {
    * @deprecated Use {@link Cluster.addAsgCapacityProvider} instead.
    */
   public addCapacity(id: string, options: AddCapacityOptions): autoscaling.AutoScalingGroup {
-    if (options.machineImage && options.machineImageType) {
-      throw new Error('You can only specify either machineImage or machineImageType, not both.');
-    }
+    // Do 2-way defaulting here: if the machineImageType is BOTTLEROCKET, pick the right AMI.
+    // Otherwise, determine the machineImageType from the given AMI.
+    const machineImage = options.machineImage ??
+      (options.machineImageType === MachineImageType.BOTTLEROCKET ? new BottleRocketImage() : new EcsOptimizedAmi());
```

Comment:
What happens now if both `machineImage` and `machineType` are specified but they don't match?

Also, given the type below in line 303, should we use `EcsOptimizedImage.amazonLinux2()`, instead, to make the connection between the two more explicit?

Reply:
> What happens now if both machineImage and machineType are specified but they don't match?

User's responsibility. I think we need this flexibility in case people have `MyCustomBottleRocketImage` for which the `instanceof` check would fail.

> Also, given the type below in line 303, should we use EcsOptimizedImage.amazonLinux2(), instead, to make the connection between the two more explicit?

Maybe. I think the code is the way it is because we're afraid of breaking existing templates.